sljit_src/sljitNativeMIPS_32.c

```diff
@@ -26,6 +26,25 @@
 
 /* mips 32-bit arch dependent functions. */
 
+static sljit_s32 emit_copysign(struct sljit_compiler *compiler, sljit_s32 op,
+		sljit_sw src1, sljit_sw src2, sljit_sw dst)
+{
+	int is_32 = (op & SLJIT_32);
+	sljit_ins fr_pair = is_32 ? 0 : (1 << 11);
+
+	if (!is_32)
+		FAIL_IF(push_inst(compiler, MFC1 | T(TMP_REG1) | FS(src1), DR(TMP_REG1)));
+	FAIL_IF(push_inst(compiler, MFC1 | T(TMP_REG2) | FS(src1) | fr_pair, DR(TMP_REG2)));
+	FAIL_IF(push_inst(compiler, MFC1 | T(TMP_REG3) | FS(src2) | fr_pair, DR(TMP_REG3)));
+	FAIL_IF(push_inst(compiler, XOR | T(TMP_REG2) | D(TMP_REG3) | S(TMP_REG3), DR(TMP_REG3)));
+	FAIL_IF(push_inst(compiler, SRL | T(TMP_REG3) | D(TMP_REG3) | SH_IMM(31), DR(TMP_REG3)));
+	FAIL_IF(push_inst(compiler, SLL | T(TMP_REG3) | D(TMP_REG3) | SH_IMM(31), DR(TMP_REG3)));
+	FAIL_IF(push_inst(compiler, XOR | T(TMP_REG3) | D(TMP_REG2) | S(TMP_REG2), DR(TMP_REG2)));
+	if (!is_32)
+		FAIL_IF(push_inst(compiler, MTC1 | T(TMP_REG1) | FS(dst), MOVABLE_INS));
+	return push_inst(compiler, MTC1 | T(TMP_REG2) | FS(dst) | fr_pair, MOVABLE_INS);
```

Comment:
I am not sure I understand this code. Copysign copies only one bit, you never need to touch the second 32 bit part for doubles.

Reply:
It is there so the same code could be used with floats; TMP_REG1 (that is needed to hold the lower int in doubles and that is otherwise copied AS-IS from src to dst) goes unused

Reply:
Ok, I get it now. We have two cases:
a) src1 == dst, when only the sign bit needs to be copied into src
b) src1 != dst, where the lower word needs to be copied as well

I think the original code only did (a), the new only (b), so we should support both.